### ISSUE TITLE
Refactor Voxel.js

### ISSUE THREAD

**Issue by kenny-designs** [kenny-designs/voxel-edit#4]
The Voxel.js module is a bit all over the place and rather monolithic. This worked fine while following the tutorial associated with the code, with thanks to [threefundamentals.org](https://threejsfundamentals.org/threejs/lessons/threejs-voxel-geometry.html), but it now must be redone to account for the needs of Voxel Edit.

The specifics of this refactor I am not sure of yet. This will likely become more clear and the Brush and ColorPalette components are created.

**Comment by kenny-designs** [comment]
So far, I've taken the old Voxel.js code and split it into several files.

Instead of a single Voxel function there is now a VoxelEditor class that handles interfacing with the 3D scene and the main render loop. This is our *driver* class of sorts and is what the VoxelManager component uses to communicate with the actual voxels.

In addition, the VoxelWorld class that handles all our voxel data is now in its own VoxelWorld.js file.

I also added a Brush class that is used by the VoxelEditor to track what the current brush is and allows for mutating that data. A bit of a warning though: there are now two Brush.js files! One for the React Brush component and one for the regular Brush class. This isn't good! Find a better name for the new Brush class or perhaps add something to the component Brush class to indicate that it is a React component.

The refactor needs a little more work as well. The code is still rather messy and needs to be cleaned up. Further work will need to be done before implement the ColorPalette component.

**Comment by kenny-designs** [comment]
Refactor seems to be complete. Now merging into dev and main.

Now we can start the ColorPalette component.
